mlir/tools/mlir-tblgen/OpFormatGen.cpp

```diff
@@ -842,7 +842,7 @@ static void genElementParserStorage(FormatElement *element, const Operator &op,
       }
     } else {
       body << "  ::mlir::OpAsmParser::UnresolvedOperand " << name
-           << "RawOperands[1];\n"
+           << "RawOperands[1] = {};\n"
            << "  ::llvm::ArrayRef<::mlir::OpAsmParser::UnresolvedOperand> "
            << name << "Operands(" << name << "RawOperands);";
```

Comment:
Seems to me that we could simplify it here:
```
     body << "  ::mlir::OpAsmParser::UnresolvedOperand " << name
            << "RawOperand;\n"
            << "  ::llvm::ArrayRef "
            << name << "Operands(&" << name << "RawOperand, 1);";
```

Something like this (I haven't tried compiling it)

Reply:
Nope :(
```
build/tools/mlir/include/mlir/Dialect/Bufferization/TransformOps/BufferizationTransformOps.cpp.inc:502:27: error: use of undeclared identifier 'targetRawOperands'; did you mean 'targetRawOperand'?
  if (parser.parseOperand(targetRawOperands[0]))
                          ^~~~~~~~~~~~~~~~~
                          targetRawOperand
```
Same for types well.

Guessing this is supposed to be an array and named in a special way for the parsing logic to be length-agnostic (not there's another branch in the codegen right above this under `if (operand->getVar()->isVariableLength())`).

Reply:
> Nope :(
> 
> ```
> build/tools/mlir/include/mlir/Dialect/Bufferization/TransformOps/BufferizationTransformOps.cpp.inc:502:27: error: use of undeclared identifier 'targetRawOperands'; did you mean 'targetRawOperand'?
>   if (parser.parseOperand(targetRawOperands[0]))
>                           ^~~~~~~~~~~~~~~~~
>                           targetRawOperand
> ```
> 
> Same for rawTypes as well.


That's because you stopped short of fixing the other occurrences of RawOperands I believe? Just remove the `[0]` and it'll work I believe.

Reply:
I don't think so: the `[0]` is not incidental there. It could also be `[1]`? [this line](https://github.com/llvm/llvm-project/pull/85631/files#diff-c15a25fdac9b602ad4f97cbd475fdb4b78e7a4b3ab741886c9dec51a78f05359R838) would break if we change the parse() generator I guess. or do you mean to have an extra branch there to handle the difference?

Reply:
my guess on this, given the error, is: the original author wanted to have the 1-vs-many case being handled in a single place (that I modify here). then, all the other places working with `*RawOperands` / `*RawTypes` won't need any special treatment of "single element" and can just use `.size()` of some generation-time variable to produce correct C++. and this is why it's an array of 1 element instead of an object?

Reply:
> I don't think so

But did you try? Because I did and `ninja check-mlir` is passing for me: https://github.com/llvm/llvm-project/commit/b44b9f71ecf51443b163e02f6bb64462cb437ae7

Reply:
no, I haven't. let me try your way then. how come the other "case" is not triggered anywhere? (the one where we have variable length operands and a `SmallVector<>` instead of a c array)